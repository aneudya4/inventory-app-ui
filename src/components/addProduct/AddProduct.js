import React from 'react';
import config from '../config';
import ProductForms from '../productsForm/ProductsForm';
import './addProduct.css';
const AddProduct = ({ userId, history, addNewProduct }) => {
  const handleSubmit = (e) => {
    e.preventDefault();
    const {
      product_name,
      product_price,
      product_description,
      product_stock,
      product_provider,
    } = e.target.elements;
    const product = {
      product_name: product_name.value,
      unit_price: parseFloat(product_price.value),
      description: product_description.value,
      stock_total: parseInt(product_stock.value),
      provider: product_provider.value,
      user_id: userId,
    };
    console.log(product, 'MMG');
    fetch(`${config.API_ENDPOINT}/products/`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json',
        Authorization: `Bearer ${config.API_KEY}`,
      },
      body: JSON.stringify(product),
    })
      .then((res) => {
        if (!res.ok) return res.json().then((e) => Promise.reject(e));
        return res.json();
      })
      .then((product) => {
        addNewProduct(product);
        history.push('/auth/dashboard/products');
      })
      .catch((error) => {
        console.error({ error });
      });
  };

  return (
    <div className='add-product'>
      <h3> Add New Product</h3>
      <form onSubmit={handleSubmit}>
        <ProductForms />
      </form>
    </div>
  );
};
export default AddProduct;
