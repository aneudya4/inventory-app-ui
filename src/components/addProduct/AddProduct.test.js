import { shallow } from 'enzyme';

import AddProduct from './AddProduct.js';

describe('Test on the <AddProduct/> Component', () => {
  it('renders AddProduct without crashing', () => {
    const wrapper = shallow(<AddProduct />);
    expect(wrapper.find('form').exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
