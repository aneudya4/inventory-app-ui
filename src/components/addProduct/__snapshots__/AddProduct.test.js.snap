// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on the <AddProduct/> Component renders AddProduct without crashing 1`] = `
<div
  className="add-product"
>
  <h3>
     Add New Product
  </h3>
  <form
    onSubmit={[Function]}
  >
    <ProductForms />
  </form>
</div>
`;
